Install some dummy and optional packages using conda-forge (micromamba)

mkoeppe opened the pull request:
<!-- Please provide a concise, informative and self-explanatory title. -->
<!-- Don't put issue numbers in the title. Put it in the Description below. -->
<!-- For example, instead of "Fixes #12345", use "Add a new method to multiply two integers" -->

### :books: Description

<!-- Describe your changes here in detail. -->
`sage -i pandoc` or `make pandoc` will now use [micromamba](https://mamba.readthedocs.io/en/latest/user_guide/micromamba.html) to install the package into an isolated conda environment in `$SAGE_LOCAL/var/lib/sage/conda`.
Likewise for the other dummy packages.

We also reduce normal optional packages `info`, `valgrind`, `rubiks` to conda-installed packages.

**This reduces platform support for these optional packages to the [platforms supported by conda-forge / micromamba](https://mamba.readthedocs.io/en/latest/installation.html#linux-and-macos), in particular, it drops support for 32-bit Linux.** See:
- https://github.com/sagemath/sage/issues/32074

This is the first step toward:
- #35583
<!-- Why is this change required? What problem does it solve? -->
<!-- If this PR resolves an open issue, please link to it here. For example "Fixes #12345". -->
<!-- If your change requires a documentation PR, please link it appropriately. -->

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. It should be `[x]` not `[x ]`. -->

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on
- #12345: short description why this is a dependency
- #34567: ...
-->

<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

Commit:
build/bin/sage-print-system-package-command: Handle system 'mamba'

Commit:
build/pkgs/{ffmpeg,git,github_cli,imagemagick,pandoc,pdf2svg,texlive}: Change from dummy package to conda package

Commit:
bootstrap: Add configure option '--enable-SPKG' for conda-installable packages

bootstrap: For conda-installed packages, also emit --enable-SPKG advice

Commit:
build/pkgs/valgrind: Change to a conda-installable package

Commit:
build/pkgs/info: Change to mamba-installed

Commit:
build/pkgs/rubiks: Change to mamba-installed

Commit:
src/doc/en/developer/packaging.rst: Mention conda packages

src/doc/en/developer/packaging.rst: Expand on conda packages

isuruf commented on src/bin/sage-env:
```diff
@@ -251,6 +251,21 @@ if [ -z "${SAGE_ORIG_PATH_SET}" ]; then
+    # https://mamba.readthedocs.io/en/latest/user_guide/micromamba.html
+    MAMBA_ROOT_PREFIX="$SAGE_LOCAL/var/lib/sage/conda"
+    CONDA_DEFAULT_ENV=sage
+    CONDA_PREFIX="$MAMBA_ROOT_PREFIX/envs/$CONDA_DEFAULT_ENV"
```
Do we need two different prefixes? `CONDA_PREFIX` can just be equal to `MAMBA_ROOT_PREFIX`

mkoeppe replied:
Thanks, I'll try this. Didn't find good documentation on the environment variables that take effect

mkoeppe replied:
Done in 7c67f69

isuruf commented on build/pkgs/conda/spkg-install:
```diff
@@ -0,0 +1,32 @@
+    PLATFORM="linux"
+    if [ "$ARCH" = "aarch64" ]; then
+        ARCH="aarch64";
+    elif [ $ARCH == "ppc64le" ]; then
```
```suggestion
    elif [ "$ARCH" = "ppc64le" ]; then
```

isuruf commented on src/bin/sage-env:
```diff
@@ -251,6 +251,19 @@ if [ -z "${SAGE_ORIG_PATH_SET}" ]; then
+        # Unset environment variables that may have come from outside
+        unset CONDA_EXE
+        unset CONDA_PYTHON_EXE
+        unset CONDA_DEFAULT_ENV
```
Setting these env variables would interfere with an external conda install which would be providing other packages for example python?

mkoeppe replied:
Good point; I'll make all settings except for PATH local to the package installation script.

mkoeppe replied:
Done in 5439fd9

Commit:
build/pkgs/conda: New optional package

src/bin/sage-env: Add the conda env to PATH if present

build/make/Makefile.in: Try to install script packages without spkg-install using mamba

build/make/Makefile.in, src/bin/sage-env: Activate the conda env for installing, remove user conda env vars

Switch from mamba-forge to micromamba

build/pkgs/conda/SPKG.rst: Update

Simplify placement of conda env

Update build/pkgs/conda/spkg-install

Co-authored-by: Isuru Fernando <isuruf@gmail.com>

src/bin/sage-env: Do not override conda env vars here; do that only in build/make/Makefile.in
